python/pyarrow/_dataset.pyx

```diff
@@ -1838,7 +1838,7 @@ cdef class FileFragment(Fragment):
             typ = ""
         partition_dict = get_partition_keys(self.partition_expression)
         partition = ", ".join(
-            [f"{key}={val}" for key, val in partition_dict.items()]
+            sorted([f"{key}={val}" for key, val in partition_dict.items()])
```

Comment:
Was this needed to get some test passing? It seems strange (or wrong?) that this needs to be sorted

Reply:
Yes, the `__repr__` test was failing. I thought it might be nice to have deterministic output for the `__repr__`. The key/value pair ordering was swapped when I compiled with Cython 3. Would it be better to update the test case to handle non-determinism?

https://github.com/apache/arrow/blob/main/python/pyarrow/tests/test_dataset.py#L1614
```
def test_fragments_repr(tempdir, dataset):
    # partitioned parquet dataset
    fragment = list(dataset.get_fragments())[0]
    assert (
        repr(fragment) ==
        "<pyarrow.dataset.ParquetFileFragment path=subdir/1/xxx/file0.parquet "
        "partition=[key=xxx, group=1]>"
    )
```

Reply:
I'll go ahead and remove sorting here and have the test handle non-deterministic repr output